partition_gpu/partition_gpu.go

```diff
@@ -98,18 +107,27 @@ func main() {
 		os.Exit(1)
 	}
 	if !migModeEnabled {
+		glog.Infof("Checking the GPU type now.")
+		gpuType, err := checkGpuType()
+		if err != nil {
+			glog.Errorf("Failed to check GPU Type: %v", err)
+			os.Exit(1)
+		}
+		glog.Infof("Got GPU type used: %s", gpuType)
 		glog.Infof("MIG mode is not enabled. Enabling now.")
 		if err := enableMigMode(); err != nil {
 			glog.Errorf("Failed to enable MIG mode: %v", err)
 			os.Exit(1)
 		}
-		glog.Infof("Rebooting node to enable MIG mode")
-		if err := rebootNode(); err != nil {
-			glog.Errorf("Failed to trigger node reboot after enabling MIG mode: %v", err)
-		}
+		if gpuType == Nvidia40gbA100 || gpuType == Nvidia80gbA100 {
```

Comment:
Add more comments for why a100 can reboot, whereas h100 doesn't reboot. And add reference: https://docs.nvidia.com/datacenter/tesla/mig-user-guide/#reset-hopper

Reply:
Done, added references